### PR TITLE
close connection

### PR DESCRIPTION
**Pull request by chungen0126**
@steinsgateted close connection

### CODE REVIEW

File: dcfs/dcfs-share/dcfs-run/demo.py [hunk 1]
```diff
@@ -122,6 +122,8 @@ def send_task_status(task_id, status, message, link='', table_name=''):
                 logging.info("Retrieving data from MySQL")
                 send_task_status(task_id, TASKSTATUS_PROCESSING, "Retrieving data from MySQL")
                 locals()['df%d'%i] = pd.read_sql(d['sql'], con=db_engine)
+                db_engine.connect().close()
```
Comment (CW-B-W) [inline dcfs/dcfs-share/dcfs-run/demo.py:125]: Hello @chungen0126,
I don't really get this.

In this line, do you mean to initiate a new connection to the DBMS, and then close the connection immediately?

ref:
https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine.connect

Comment (CW-B-W) [reply dcfs/dcfs-share/dcfs-run/demo.py:125]: I think only 
```suggestion
                db_engine.dispose()
```
is needed?

ref: https://stackoverflow.com/a/47964066/16113175

File: web/flask/hello.py [hunk 2]
```diff
@@ -79,14 +79,17 @@ def mongodb_list_all_dbs(username, password, ip, port):
...
+    return listdb
 
 def mongodb_list_all_tables(username, password, ip, port, db_name):
     if username != '':
         mongo_client = MongoClient(f'mongodb://{username}:{password}@{ip}:{port}/')
     else:
         mongo_client = MongoClient(f'mongodb://{ip}:{port}/')
     db = mongo_client[db_name]
+    mongo_client.close()
     return sorted(db.list_collection_names())
```
Comment (CW-B-W) [inline web/flask/hello.py:93]: Hello @chungen0126 ,
Does
```python
return sorted(db.list_collection_names())
```
work when we have done
```python
mongo_client.close()
```
?

File: web/flask/hello.py [hunk 3]
```diff
@@ -228,11 +243,13 @@ def phoenix_list_all_types(table_name, ip='hbase-master', port='8765'):
...
     for item in res:
         d[item['COLUMN_NAME']] = TYPE_MAP[item['DATA_TYPE']]
+    conn.close()
     return d
 
 def phoenix_drop_table(table_name, ip='hbase-master', port='8765'):
     conn = phoenixdb.connect('http://%s:%s' % (ip, port))
     cursor = conn.cursor(cursor_factory=phoenixdb.cursor.DictCursor)
+    conn.close()
     cursor.execute("DROP TABLE {}".format(table_name))
```
Comment (steinsgateted) [inline web/flask/hello.py:253]: same meaning as 252 and 253

**Review by steinsgateted — CHANGES_REQUESTED** [review]

File: web/flask/requirements.txt [hunk 4]
```diff
@@ -15,4 +15,5 @@ pymongo==3.12.1
 pymssql==2.2.2
 PyMySQL==1.0.2
 SQLAlchemy==1.4.27
-xlrd==1.2.0
\ No newline at end of file
+xlrd==1.2.0
+protobuf==3.20.0
```
Comment (steinsgateted) [inline web/flask/requirements.txt:19]: prevent forgetting:
Fixed the protobuf version because the flask web page was not opened

Comment (CW-B-W) [reply web/flask/requirements.txt:19]: Hello @steinsgateted ,
What happens if we don't specify the `protobuf==3.20.0` version?
(just out of curiosity)

Comment (chungen0126) [reply web/flask/requirements.txt:19]: > What happens if we don't specify the protobuf==3.20.0 version?

flask  will crash.

Comment (CW-B-W) [reply web/flask/requirements.txt:19]: Why?

Comment (steinsgateted) [reply web/flask/requirements.txt:19]: I have no idea.

**Review by steinsgateted — APPROVED** [review]
+1 Thanks.

File: web/flask/requirements.txt (hunk 4 above)
Comment (chungen0126) [reply web/flask/requirements.txt:19]: There must be some Compatibility issues, I think specify the protobuf version is more stable.

Comment (CW-B-W) [reply web/flask/requirements.txt:19]: Thanks

**Pull request merged**